compound_disease/compound_treats_disease/transfer_learning_experiment/transfer_learning.py

```diff
@@ -0,0 +1,270 @@
+    for key in spreadsheet_names
+}
+
+for key in candidate_dfs:
+    print("Size of {} set: {}".format(key, candidate_dfs[key].shape[0]))
+
+
+# In[ ]:
+
+
+label_functions = (
+    list(CD_LFS['CtD_DB'].values()) + 
+    list(CD_LFS['CtD_TEXT'].values()) + 
+    list(CD_LFS['CD_BICLUSTER'].values()) + 
+    list(CG_LFS["CbG_TEXT"].values())+
+    list(DG_LFS["DaG_TEXT"].values())
+) 
+
+if quick_load:
+    label_matricies = pickle.load(open("data/label_matricies.pkl", "rb"))
+
+else:
+    label_matricies = {
+        key:label_candidates(
+            session,
+            candidates_dfs[key]['candidate_id'],
+            label_functions,
+            num_threads=10,
+            batch_size=candidates_dfs[key]['candidate_id'].shape[0]
+        )
+        for key in candidate_dfs
+    }
+    
+    pickle.dump(label_matricies, open("data/label_matricies.pkl", "wb"))
+
+
+# In[ ]:
+
+
+lf_names = [
+    label_matricies['test'].get_key(session, index).name 
+    for index in range(label_matricies['test'].shape[1])
+]
+
+
+# ## Observe Label Function Properties
+
+# Before training the generative model, it is a good idea to observe label functions and their output. In the cells below we take a look at generated heatmaps and observe the underlying structure.
+
+# In[ ]:
+
+
+plot_label_matrix_heatmap(label_matricies['train'].T, 
+                          yaxis_tick_labels=lf_names, 
+                          figsize=(10,13))
+
+
+# # Train Generative Model
+
+# In[ ]:
+
+
+cd_db = get_columns(session, label_matricies['train'], CD_LFS, "CtD_DB")
+cd_text = get_columns(session, label_matricies['train'], CD_LFS, "CtD_TEXT")
+cd_bicluster = get_columns(session, label_matricies['train'], CD_LFS, "CD_BICLUSTER")
+cg_text = get_columns(session, label_matricies['train'], CG_LFS, "CbG_TEXT")
+dg_text = get_columns(session, label_matricies['train'], DG_LFS, "DaG_TEXT")
+
+
+# In[ ]:
+
+
+# Dictionary specifying the different models 
+# for this analysis
+model_dict = {
+    "CtD_DB": cd_db,
+    "CtD_TEXT": cd_text,
+    "CD_BICLUSTER": cd_bicluster,
+    "CbG_TEXT": cg_text,
+    "DaG_TEXT": dg_text,
+    "CtD_DB_TEXT": cd_db+cd_text,
+    "CtD_ALL": cd_db+cd_text+cd_bicluster,
+    "All_the_jawns": cd_db+cd_text+cd_bicluster+cg_text+dg_text
+}   
+
+
+# In[ ]:
+
+
+L = label_matricies['train'].toarray()
+L[L < 0] = 2
+L_dev = label_matricies['dev'].toarray()
+L_dev[L_dev < 0] = 2
+L_test = label_matricies['test'].toarray()
+L_test[L_test < 0] = 2
+
+label_model = LabelModel(k=2, seed=100)
+
+
+# In[ ]:
+
+
+reg_param_grid = pd.np.round(pd.np.linspace(1e-1, 1, num=30), 3)
+grid_results = defaultdict(dict)
+for model in tqdm_notebook(model_dict):
+    for reg_param in reg_param_grid:
+        label_model.train(L[:, model_dict[model]], n_epochs=1000, verbose=False, lr=0.01, l2=reg_param)
+        grid_results[model][str(reg_param)] = label_model.predict_proba(L_dev[:, model_dict[model]])[:,0]
+
+
+# In[ ]:
+
+
+for model in grid_results:
```

Comment:
You're dev is your holdout set from your training?  For your dev set you have labels and you're using the performance on the dev set to select your hyperparams?

There's not CV here correct?

Reply:
Correct in this context the training set is unlabeled which means one cannot use CV without significant amount of effort to get gold labels. In this case I am using a dev set to select a model.